src/main/cpp/tests/timezones.cpp

```diff
@@ -0,0 +1,280 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "timezones.hpp"
+
+#include <climits>
+
+#include <cudf/wrappers/timestamps.hpp>
+
+#include <cudf_test/base_fixture.hpp>
+#include <cudf_test/column_utilities.hpp>
+#include <cudf_test/column_wrapper.hpp>
+#include <cudf_test/iterator_utilities.hpp>
+#include <cudf_test/type_lists.hpp>
+
+using seconds_col =
+    cudf::test::fixed_width_column_wrapper<cudf::timestamp_s, cudf::timestamp_s::rep>;
+
+using millis_col =
+    cudf::test::fixed_width_column_wrapper<cudf::timestamp_ms, cudf::timestamp_s::rep>;
+
+using micros_col =
+    cudf::test::fixed_width_column_wrapper<cudf::timestamp_us, cudf::timestamp_s::rep>;
+
+class TimeZoneTest : public cudf::test::BaseFixture {
+protected:
+  void SetUp() override
+  {
+    transitions = make_transitions_table();
+  }
+  std::unique_ptr<cudf::table> transitions;
+
+private:
+  std::unique_ptr<cudf::table> make_transitions_table()
+  {
+    auto instants_from_utc_col = cudf::test::fixed_width_column_wrapper<int64_t>(
+        {LONG_MIN, LONG_MIN, -1585904400L, -933667200L, -922093200L, -908870400L, -888829200L,
+         -650019600L, 515527200L, 558464400L, 684867600L});
+    auto instants_to_utc_col = cudf::test::fixed_width_column_wrapper<int64_t>(
+        {LONG_MIN, LONG_MIN, -1585904400L, -933634800L, -922064400L, -908838000L, -888796800L,
+         -649990800L, 515559600L, 558493200L, 684896400L});
+    auto utc_offsets_col = cudf::test::fixed_width_column_wrapper<int32_t>(
+        {18000, 29143, 28800, 32400, 28800, 32400, 28800, 28800, 32400, 28800, 28800});
+    auto struct_column = cudf::test::structs_column_wrapper{
+        {instants_from_utc_col, instants_to_utc_col, utc_offsets_col},
+        {1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1}};
```

Comment:
There's a shortcut for the "all valid" case, in [`cudf_test/iterator_utilities.hpp`](https://github.com/rapidsai/cudf/blob/branch-23.12/cpp/include/cudf_test/iterator_utilities.hpp#L116):
```c++
    auto struct_column = cudf::test::structs_column_wrapper{
￼        {instants_from_utc_col, instants_to_utc_col, utc_offsets_col},
￼        cudf::test::iterators::no_nulls()};
```
Please feel free to ignore, if the current version reads better for you.

Reply:
Please mark all variables as `const` whenever possible.